hw/cxl/cxl-host.c

```diff
@@ -210,36 +209,36 @@ static MemTxResult cxl_read_cfmws(void *opaque, hwaddr addr, uint64_t *data,
 
     d = cxl_cfmws_find_device(fw, addr);
     if (d == NULL) {
+        trace_cxl_debug_message("CXL device not found");
         *data = 0;
         /* Reads to invalid address return poison */
         return result;
     }
-    type = object_get_typename(OBJECT(d));
 
-    g_assert(addr < 0x8000000);
-
-    if (g_strcmp0(type, "cxl-type1") == 0)
-        result = cxl_type1_read(d, addr + fw->base, data, size, attrs);
-    else if (g_strcmp0(type, "cxl-type2") == 0)
-        result =
-            cxl_host_type2_hcoh_read(d, addr + fw->base, data, size, attrs);
-    else if (g_strcmp0(type, "cxl-type3") == 0) {
-        if (cxl_is_remote_root_port(d)) {
+    if (cxl_is_remote_root_port(d)) {
```

Comment:
Could you explain why this check needs to be added? How did it work without it before?